### PR TITLE
ci: use caching to speed up workflows

### PR DESCRIPTION
**Pull request by EvadeMaster**
# 🤖 Speed up workflows
Enable GitHub cache to speed up workflows 

## 👋 Resolve
resolve: #611

## Additional...
<details>
<summary>Please don't look at the past Please don't look at the past Please don't look at the past Please don't look at the past Please don't look at the past</summary>
https://user-images.githubusercontent.com/93124920/227767135-bc7909db-6b63-4541-860d-fec490925b0f.png
</details>

### CODE REVIEW

**Review by PalmDevs — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,16 +33,17 @@ jobs:
...
         run: flutter pub get
       - name: Generate files with Builder
         run: flutter packages pub run build_runner build --delete-conflicting-outputs
       - name: Build with Flutter
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          SIGNING_KEY_ALIAS: ${{ secrets.SIGNING_KEY_ALIAS }}
-          SIGNING_KEY_PASSWORD: ${{ secrets.SIGNING_KEY_PASSWORD }}
-          SIGNING_STORE_PASSWORD: ${{ secrets.SIGNING_KEYSTORE_PASSWORD }}
+          #SIGNING_KEY_ALIAS: ${{ secrets.SIGNING_KEY_ALIAS }}
```
Comment (PalmDevs) [inline .github/workflows/release.yml:44]: Does this step not require the keystore credentials?

Comment (EvadeMaster) [reply .github/workflows/release.yml]: Whoops, the revert might have lost during the squash, I'll go ahead and fix it

**Review by oSumAtrIX — APPROVED** [review]

**Comment by PalmDevs** [comment]
@EvadeMaster You did not need to revert that! It works perfectly fine without those. See my [test](https://github.com/PalmDevs/revanced-manager/actions/runs/4524559562/jobs/7968627317).

**Comment by Ushie** [comment]
> @EvadeMaster You did not need to revert that! It works perfectly fine without those. See my [test](https://github.com/PalmDevs/revanced-manager/actions/runs/4524559562/jobs/7968627317).

Flutter can build APKs that aren't signed

**Comment by PalmDevs** [comment]
> > @EvadeMaster You did not need to revert that! It works perfectly fine without those. See my [test](https://github.com/PalmDevs/revanced-manager/actions/runs/4524559562/jobs/7968627317).
> 
> Flutter can build APKs that aren't signed, but we do need them to be signed

It's signed in the next step. Look at the workflow file.

**Comment by EvadeMaster** [comment]
> > > @EvadeMaster You did not need to revert that! It works perfectly fine without those. See my [test](https://github.com/PalmDevs/revanced-manager/actions/runs/4524559562/jobs/7968627317).
> > 
> > 
> > Flutter can build APKs that aren't signed, but we do need them to be signed
> 
> It's signed in the next step. Look at the workflow file.

I believe, you might’ve incorrectly configured your workflow, because ilharp/sign-android-release has these parameters to be filled out; otherwise, it will fail due to being unable to sign an application without key.
> signingKey
> keyAlias
> keyStorePassword


Check your configuration:
https://github.com/revanced/revanced-manager/blob/a79f883a0f8adc5f285a43d1203697b36e81afd8/.github/workflows/release.yml#L48-L56

**Comment by PalmDevs** [comment]
> I believe, you might’ve incorrectly configured your workflow, because `ilharp/sign-android-release` has these parameters to be filled out; otherwise, it will fail due to being unable to sign an application without key.
> > signingKey
> > keyAlias
> > keyStorePassword

No? I've filled them and generated my own keystore. See [my workflow configuration](https://github.com/PalmDevs/revanced-manager/blob/dev/.github/workflows/release.yml#L44).

I've also filled them accordingly.

**Pull request merged**